TestFoundation/TestCodable.swift

```diff
@@ -33,25 +33,28 @@ private func makePersonNameComponents(namePrefix: String? = nil,
     return result
 }
 
-func expectRoundTripEquality<T : Codable>(of value: T, encode: (T) throws -> Data, decode: (Data) throws -> T) where T : Equatable {
+func expectRoundTripEquality<T : Codable>(of value: T, encode: (T) throws -> Data, decode: (Data) throws -> T) -> Bool where T : Equatable  {
     let data: Data
     do {
         data = try encode(value)
     } catch {
-        fatalError("Unable to encode \(T.self) <\(value)>: \(error)")
+        XCTFail("Unable to encode \(T.self) <\(value)>: \(error)")
+        return false
```

Comment:
Personally I would prefer not to have a return value of false which may forget to be called by the calling site. Instead, you should pass in the message into the `expect()` call and then there's no need to have the `XCTAssertTrue` sprinkled throughout the code base.

In other words, I like the idea of fatalError -> XCTFail change, but I don't like the modification of return type to Bool or the other changes in this file.

Reply:
If the return value isn't acted upon then a compiler warning is emitted.

The advantage of having the `XCTAssert..` in the `test_()` function is that failing tests show up more easily when running the tests under Xcode, when run as normal unit tests and not via `TestFoundation`. This makes it quicker to know what is failing as Xcode will highlight the failing test rather than having to step though the code.

Reply:
If we're going to do something which requires the caller take action (as opposed to merely a warning when it doesn't work) then we should change this from returning a Bool to making it a throw, which will explicitly force unadorned calls to be handled. What do you think?

Reply:
Yes, agreed a `throw` is better since it causes and error rather than a warning if not handled. I will add a commit to see what it looks like.